src/networkManager.coffee

```
@@ -0,0 +1,122 @@
+Promise = require 'bluebird'
+DBus = require './dbus-promise'
+_ = require 'lodash'
+
+dbus = new DBus()
+
+bus = dbus.getBus('system')
+
+systemd = require './systemd'
+
+SERVICE = 'org.freedesktop.NetworkManager'
+WHITE_LIST = ['resin-vpn', 'eth0']
+
+NM_STATE_CONNECTED_GLOBAL = 70
+NM_DEVICE_TYPE_WIFI = 2
+NM_CONNECTIVITY_FULL = 5
+
+exports.start = ->
+	systemd.start('NetworkManager.service')
+
+exports.stop = ->
+	systemd.stop('NetworkManager.service')
+
+exports.isSetup = ->
+	getConnections()
+	.map(validateConnection)
+	.then (results) ->
+		return false in results
+
+exports.setCredentials = (ssid, passphrase) ->
+	connection = {
+		'802-11-wireless': {
+			ssid: _.invokeMap(ssid, 'charCodeAt')
+		},
+		connection: {
+			id: ssid,
+			type: '802-11-wireless',
+		},
+		'802-11-wireless-security': {
+			'auth-alg': 'open',
+			'key-mgmt': 'wpa-psk',
+			'psk': passphrase,
+		}
+	}
+
+	console.log('Saving connection')
+	console.log(connection)
+
+	bus.getInterfaceAsync(SERVICE, '/org/freedesktop/NetworkManager/Settings', 'org.freedesktop.NetworkManager.Settings')
+	.then (settings) ->
+		settings.AddConnectionAsync(connection)
+
+exports.clearCredentials = ->
+	getConnections()
+	.map(deleteConnection)
+
+exports.connect  = (timeout) ->
+	getDevices()
+	.filter (device) ->
+		validateDevice(device)
+	.then (validDevices) ->
+		if validDevices.length is 0
+			reject()
```

This should be a throw, `reject` is undefined here (also because there's no return it'd still try to run the code below regardless)